tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_controller_component.ts

```diff
@@ -112,26 +122,62 @@ export class LinkedTimeFobControllerComponent {
     }
   }
 
-  isDraggingLower(position: number, movement: number): boolean {
+  isDraggingLower(event: MouseEvent): boolean {
+    let position = this.fobCardData.histograms ? event.clientY : event.clientX;
+    let movement = this.fobCardData.histograms
+      ? event.movementY
+      : event.movementX;
     return (
-      position < this.getDraggingFobTop() &&
+      position < this.getDraggingFobCenter() &&
       movement < 0 &&
-      this.getDraggingFobStep() > this.steps[0]
+      this.getDraggingFobStep() > this.getMinStep()
     );
   }
 
-  isDraggingHigher(position: number, movement: number): boolean {
+  isDraggingHigher(event: MouseEvent): boolean {
+    let position = this.fobCardData.histograms ? event.clientY : event.clientX;
+    let movement = this.fobCardData.histograms
+      ? event.movementY
+      : event.movementX;
     return (
-      position > this.getDraggingFobTop() &&
+      position > this.getDraggingFobCenter() &&
       movement > 0 &&
-      this.getDraggingFobStep() < this.steps[this.steps.length - 1]
+      this.getDraggingFobStep() < this.getMaxStep()
     );
   }
 
-  getDraggingFobTop(): number {
-    return this.currentDraggingFob !== Fob.END
-      ? this.startFobWrapper.nativeElement.getBoundingClientRect().top
-      : this.endFobWrapper.nativeElement.getBoundingClientRect().top;
+  getMaxStep(): number {
+    if (this.fobCardData.scalars) {
+      let minMax = this.fobCardData.scalars.minMax;
+      return minMax[0] < minMax[1] ? minMax[1] : minMax[0];
+    }
+
+    return this.fobCardData.histograms!.steps[
+      this.fobCardData.histograms!.steps.length - 1
+    ];
+  }
+
+  getMinStep(): number {
+    if (this.fobCardData.scalars) {
+      let minMax = this.fobCardData.scalars.minMax;
+      return minMax[0] < minMax[1] ? minMax[0] : minMax[1];
+    }
+
+    return this.fobCardData.histograms!.steps[0];
+  }
+
+  getDraggingFobCenter(): number {
+    // the fob is centered around the top when in the vertical direction and
```

Comment:
I'm actually a bit confused. Top and center don't sound the same to me. If you mean the center and the top/left is roughly at the same position. Could we just say `getDraggingFobTopOrLeft`?